### ISSUE TITLE
ApacheKiller DoS - Range Header Too Man Fields

### ISSUE THREAD

**Issue by mailtoyogeshpatel** [SpiderLabs/owasp-modsecurity-crs#229]
In our scenario we could not determine which document has a problem and blocked by mod security.
I can only say that we are opening pdf file from browser.I am not sure which type of documents has such bytes range which violets the rule.
How to identify such range and what i need to change in rule to allow more range.

**Comment by mailtoyogeshpatel** [comment]
In our application we are getting "ApacheKiller DoS - Range Header Too many fields" in modsecurity log.

      As we are unable to recognisze the same, can you please tell us what could be the root cause  and what action should be taken to resolve this issue.

 See the attached image.

![dos_attack](https://cloud.githubusercontent.com/assets/11291299/7704055/a8e7ca48-fe59-11e4-8a28-709ee9ebdb64.jpg)

**Comment by csanders-git** [comment]
It'd be nice to have an example document. We are working to improve these rules and it is good to know browsers may split things up like this, although it is pretty unclear why in this case there are so many range fields.

**Comment by marcstern** [comment]
This is due to a PDF plugin performing (pseudo) streaming - displaying pages before the whole PDF is downloaded.

**Comment by dune73** [comment]
The Core Rule Set 3 has moved this rule out of the default installation. If you raise the Paranoia Level setting to 2, the rule is back, but with a limit of 34 (!) ranges for the range header (that is the maximum we have seen in the wild by legitimate clients). In Paranoia Level 4, the limit of 5 ranges is enforced for PDFs again.

Closing this.
